scripts/utils/bundle-dts.ts

```diff
@@ -1,9 +1,21 @@
-import { generateDtsBundle } from 'dts-bundle-generator/dist/bundle-generator.js';
+import { EntryPointConfig, generateDtsBundle, OutputOptions } from 'dts-bundle-generator/dist/bundle-generator.js';
 import fs from 'fs-extra';
 
 import { BuildOptions } from './options';
 
-export async function bundleDts(opts: BuildOptions, inputFile: string) {
+/**
+ * A thin wrapper for `dts-bundle-generator` which uses our build options to
+ * set a few things up
+ *
+ * **Note**: this file caches its output to disk, and will return any
+ * previously cached file if not in a prod environment!
+ *
+ * @param opts an object holding information about the current build of Stencil
+ * @param inputFile the path to the file which should be bundled
+ * @param ouputOptions options for bundling the file
+ * @returns a string containing the bundled typedef
+ */
+export async function bundleDts(opts: BuildOptions, inputFile: string, ouputOptions?: OutputOptions): Promise<string> {
```

Comment:
I added the `outputOptions` optional parameter here because for bundling up `stencil-public-docs.d.ts` I wanted to be able to pass `noBanner: true` since we do our own banner already:

https://github.com/ionic-team/stencil/blob/43cfc584c8fbcb77122cbf2a813b1a0303085dce/src/compiler/docs/json/index.ts#L18-L26

and also I want to pass `exportReferencedTypes: false` to make sure that the inlined types aren't exported (this is the default behavior)